core/schema/modulesource.go

```diff
@@ -213,16 +214,25 @@ type buildkitClient interface {
 // - if not, try to isolate root of git repo from the ref
 // - if nothing worked, fallback as local ref, as before
 func parseRefString(ctx context.Context, bk buildkitClient, refString string) parsedRefString {
+	ctx, span := core.Tracer(ctx).Start(ctx, fmt.Sprintf("parseRefString: %s", refString))
+
+	defer span.End()
 	localParsed := parsedRefString{
 		modPath: refString,
 		kind:    core.ModuleSourceKindLocal,
 		scheme:  core.NoScheme,
 	}
 
+	if strings.HasPrefix(refString, ".") {
+		return localParsed
+	}
+
 	// First, we stat ref in case the mod path github.com/username is a local directory
 	stat, err := bk.StatCallerHostPath(ctx, refString, false)
 	if err == nil && stat.IsDir() {
 		return localParsed
+	} else if err != nil {
+		bklog.GetLogger(ctx).Errorf("parseRefString stat:%v err:%v", stat, err)
```

Comment:
I think we want to use `slog` here?

Also, I think since this is "expected", this should be lowered to a `Debug` level log.